### PR TITLE
Mint Events

### PR DESCRIPTION
**Pull request by artwoo0**
- emit a Mint Event for mint, mintWithN and multiMintWithN
- added unit tests

### CODE REVIEW

File: test/unit/NPass.test.ts [hunk 1]
```diff
@@ -289,4 +289,30 @@ describe("NPass", function () {
       ).to.be.revertedWith("NPass:INVALID_PRICE");
     });
   });
+
+  describe('Events', async () => {
+    it("Mint should emit a Mint Event", async () => {
+      const tx = await deployer.NDerivative.mint(9999);
+      const receipt = await tx.wait();
+      const mintEvent = receipt.events?.filter(e => e.event == "Mint");
+      expect(mintEvent).to.be.length(1);
```
Comment (ghost) [inline test/unit/NPass.test.ts]: can you please use the event matcher 

```suggestion
await expect(deployer.NDerivative.mint())
	.to.emit(deployer.NDerivative, 'Minted')
	.withArgs(deployer.address, tokenId);
```

File: contracts/core/NPassCore.sol [hunk 2]
```diff
@@ -24,6 +24,8 @@ abstract contract NPassCore is ERC721Enumerable, ReentrancyGuard, Ownable {
     uint256 public immutable priceForNHoldersInWei;
     uint256 public immutable priceForOpenMintInWei;
 
+    event Mint(address to, uint256 tokenId);
```
Comment (ghost) [inline contracts/core/NPassCore.sol]: Past tense is better for events (as they have already happened)

**Review by ghost — CHANGES_REQUESTED** [review]

File: contracts/core/NPassCore.sol (hunk 2 above)
Comment (artwoo0) [reply contracts/core/NPassCore.sol]: Open Zeppelin uses present tense, but past tense makes sense.

**Review by ghost — APPROVED** [review]

File: contracts/core/NPassCore.sol (hunk 2 above)
Comment (ghost) [reply contracts/core/NPassCore.sol]: Yep, that's common in Solidity but I am fighting a war to follow 10 years of Event Sourcing conventions

**Pull request merged**